packages/unimodules-test-core/android/src/main/java/expo/modules/test/core/ModuleMockInvocationHandler.kt

```diff
@@ -0,0 +1,123 @@
+package expo.modules.test.core
+
+import com.facebook.react.bridge.JavaOnlyArray
+import com.facebook.react.bridge.ReadableArray
+import expo.modules.kotlin.ModuleHolder
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.records.Record
+import expo.modules.kotlin.records.toJSMap
+import java.lang.reflect.InvocationHandler
+import java.lang.reflect.Method
+import kotlin.reflect.KClass
+
+/**
+ * The promise rejection will be transformed into this exception..
```

Comment:
~I guess `convert` might better describe the process happening here 🤔~
I've tried to find the actual differences between words `transform` and `convert`, but these could be used interchangeably in most cases (https://diffsense.com/diff/convert/transform) 😅 
```suggestion
 * The promise rejection will be converted into this exception.
```
Up to you if you accept this suggestion 👀